src/app/services/constraintService.spec.ts

```diff
@@ -77,23 +128,86 @@ function evaluateNodeConstraintWithTwoRemovalCriteria() {
+      spyOn(dataService, 'isCompleted').and.returnValue(true);
+      expect(service.evaluateCriterias(criterias)).toEqual(true);
+    });
+    it('should evaluate criterias when it is passed multiple criterias and one is false', () => {
+      const criterias = [criteria1, criteria2];
+      spyOn(dataService, 'isCompleted')
+        .withArgs(nodeId1)
+        .and.returnValue(true)
+        .withArgs(nodeId2)
+        .and.returnValue(false);
+      expect(service.evaluateCriterias(criterias)).toEqual(false);
+    });
   });
-  it('should evaluate criterias when it is passed one criteria that is true', () => {
-    const criterias = [criteria1];
-    spyOn(dataService, 'isCompleted').and.returnValue(true);
-    expect(service.evaluateCriterias(criterias)).toEqual(true);
+}
+
+function getConstraintsThatAffectNode() {
+  const constraint1: Constraint = new Constraint({
+    id: 'constraint1',
+    action: 'makeAllNodesAfterThisNotVisible',
+    targetId: nodeId5
   });
-  it('should evaluate criterias when it is passed multiple criterias and one is false', () => {
-    const criterias = [criteria1, criteria2];
-    spyOn(dataService, 'isCompleted')
-      .withArgs('node1')
-      .and.returnValue(true)
-      .withArgs('node2')
-      .and.returnValue(false);
-    expect(service.evaluateCriterias(criterias)).toEqual(false);
+  const constraint2: Constraint = new Constraint({
+    id: 'constraint2',
+    action: 'makeThisNodeNotVisitable',
+    targetId: nodeId2
+  });
+  const constraint3: Constraint = new Constraint({
+    id: 'constraint3',
+    action: 'makeThisNodeNotVisitable',
+    targetId: groupId2
+  });
+  describe('getConstraintsThatAffectNode()', () => {
+    beforeEach(() => {
+      spyOn(configService, 'getConfigParam').and.returnValue(true);
+      service.activeConstraints = [constraint1, constraint2, constraint3];
+    });
+    it(`should get the constraints that affect the node when there are no constraints that affect
+        the node`, () => {
+      getConstraintsThatAffectNodeAndExpect(nodeId1, []);
+    });
+    it(`should get the constraints that affect the node when there is a node after constraint that
+        affects the node`, () => {
+      getConstraintsThatAffectNodeAndExpect(nodeId6, [constraint1]);
+    });
+    it(`should get the constraints that affect the node when the node is the target`, () => {
+      getConstraintsThatAffectNodeAndExpect(nodeId2, [constraint2]);
+    });
+    it(`should get the constraints that affect the node when the node is a child of the
+        target`, () => {
+      getConstraintsThatAffectNodeAndExpect(nodeId4, [constraint3]);
+    });
+  });
+}
+
+function getConstraintsThatAffectNodeAndExpect(nodeId: string, expectedConstraints: Constraint[]) {
+  expect(service.getConstraintsThatAffectNode({ id: nodeId })).toEqual(expectedConstraints);
+}
+
+function orderConstraints() {
+  describe('orderConstraints()', () => {
+    it('should order constraints', () => {
```

Comment:
I think we should say what we expect the function to return to make the test clearer
```suggestion
    it('should order constraints in the order that they are in the unit', () => {
```